tree/checkMirrorImage.js

```diff
@@ -0,0 +1,20 @@
+const BinaryTree = require("./BinaryTree");
+
+let checkMirrorImage = (tree1, tree2) => {
+    let isMirrorImage = true;
+    checkIfMirrorImage = (tree1, tree2)=> {
+        isMirrorImage = isMirrorImage && ((tree1 && tree2) || (!tree1 && !tree2));
```

Comment:
did not understand this statement
```javascript
((tree1 && tree2) || (!tree1 && !tree2))
```
tree1 && tree2 . this means tree 1 exists or tree1 and tree2 both don't exist. Is my understanding correct?
If so, please create variables for these checks whose name could explain what is being done.
Also, I don't understand why would one need a global variable outside functions scope to do this.

Reply:
yes it means either both child exist or none.  
value of isMirrorImage if once set to false, we would not need to evaluate tree1 & tree2 values. if i create variables then i will have to change code to 
if(isMirrorImage) {
   let bothChildExist = tree1 && tree2;
   let bothChildNotExist = !tree1 && !tree2;
   isMirrorImage = bothChildExist || bothChildNotExist;
}

Reply:
well, without variables it looks difficult to understand. Code ko man ki tarah saaf rakho